src/main/java/duke/functionality/Ui.java

```diff
@@ -0,0 +1,158 @@
+package duke.functionality;
+
+import duke.tasks.Task;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+import java.util.List;
+
+/**
+ * Creates an Ui system that is responsible for interacting with the user.
+ */
+public class Ui {
+    private static final String BORDER = "____________________________________________________________";
+    private static final String NEW_LINE = System.lineSeparator();
+    private static final String PLACEHOLDER = "%text%";
+    private static final String DONE_TASK = "Nice! I've marked this task as done:";
+    private static final String MAX_TASK = "Sorry! You have max number of tasks stored already.";
+    private static final String ADDED_TASK = "Got it. I've added this task:";
+    private static final String REMAINING_TASK_NUM = String.format("Now you have %s tasks in the list.", PLACEHOLDER);
+    private static final String DELETED_TASK = "Noted. I've removed this task:";
+    private static final String MISSING_TASK = "There is no task at the specified index.";
+    private static final String NO_MATCHING_TASKS = "There are no matching tasks found.";
+    private static final String UNKNOWN_COMMAND = "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
```

Comment:
Perhaps it would be better to prefix certain names to constants that are related to exception reporting, user interaction and formatting.

This could help you maintain a large number of constants as your project scales in size.